src/encoder.rs

```diff
@@ -162,7 +162,7 @@ fn encode_bson<W: Write + ?Sized>(writer: &mut W, key: &str, val: &Bson) -> Enco
             try!(writer.write_u8(From::from(subtype)));
             writer.write_all(data).map_err(From::from)
         },
-        &Bson::UtcDatetime(ref v) => write_i64(writer, v.timestamp()),
+        &Bson::UtcDatetime(ref v) => write_i64(writer, v.timestamp() * 1000),
```

Comment:
Yup, for methods implemented through traits, the trait needs to be imported. The outer parens are unnecessary, but that's a small nit :)